assets/eip-5058/test/test.ts

```diff
@@ -0,0 +1,157 @@
+import "@nomiclabs/hardhat-ethers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { EIP5058Mock } from "typechain-types";
+
+describe("EIP5058 contract", function() {
+  let owner: SignerWithAddress;
+  let alice: SignerWithAddress;
+  let EIP5058: EIP5058Mock;
+  
+  beforeEach(async () => {
+    [owner, alice] = await ethers.getSigners();
+    
+    const EIP5058Factory = await ethers.getContractFactory("EIP5058Mock");
+    
+    EIP5058 = await EIP5058Factory.deploy("Mock", "M");
+  });
+  
+  it("Deployment should assign the total supply of tokens to the owner", async function() {
+    const ownerBalance = await EIP5058.balanceOf(owner.address);
+    expect(await EIP5058.totalSupply()).to.equal(ownerBalance);
+  });
+  
+  it("lockMint works", async function() {
+    const NFTId = 0;
+    const block = await ethers.provider.getBlockNumber();
+    const blockBefore = await ethers.provider.getBlock(block);
+    const timestamp = blockBefore.timestamp;
+    await EIP5058.lockMint(alice.address, NFTId, timestamp + 2);
+    
+    expect(await EIP5058.isLocked(NFTId)).eq(true);
+    expect(await EIP5058.lockerOf(NFTId)).eq(owner.address);
```

Comment:
https://github.com/ethereum/EIPs/pull/5058/files#diff-8126a15429811211509c50043cbc3d5929c79ce8c919d0078ac7c5688abc5fa4R157

When you see here, it is called via `_lock(address(0), tokenId, expired);`. In this case, operator is set to 0, but the locker is set to _msgSender(). Is this your intention?